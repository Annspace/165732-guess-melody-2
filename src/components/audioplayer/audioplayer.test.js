import React from 'react';
import renderer from 'react-test-renderer';
import AudioPlayer from "./audioplayer.jsx";
import mockData from "../../mocks/questions";

function createNodeMock(element) {
  if (element.type === `audio`) {
    // This is your fake DOM node for <p>.
    // Feel free to add any stub methods, e.g. focus() or any
    // other methods necessary to prevent crashes in your components.
    return {
      createRef() {}
    };
  }
  // You can return any object from this method for any type of DOM component.
  // React will use it as a ref instead of a DOM node when snapshot testing.
  return null;
}


it(`renders correctly`, () => {
  const options = {createNodeMock};
  const onClickTrackButtonHandler = jest.fn();
  const AudioPlayerComponent = renderer
    .create(<AudioPlayer
      onClickTrackButton={onClickTrackButtonHandler}
      playing={true}
      src={mockData.questionsData[0].song.src}/>, options).toJSON();
  expect(AudioPlayerComponent).toMatchSnapshot();
});
