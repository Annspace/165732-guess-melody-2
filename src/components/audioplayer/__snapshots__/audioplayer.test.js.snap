// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <button
    className="track__button track__button--pause"
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio
      src="https://es31-server.appspot.com/guess-melody/static/music/Whaling_City.mp3"
    />
  </div>,
]
`;
